src/gt4py/cartesian/gtscript.py

```diff
@@ -694,10 +694,23 @@ def __getitem__(self, field_spec):
         return _FieldDescriptor(dtype, axes, data_dims)
 
 
+class _ReadOnlyFieldDescriptorMaker(_FieldDescriptorMaker):
+    def __getitem__(self, field_spec):
+        if not isinstance(field_spec, collections.abc.Collection) and not len(field_spec) == 2:
+            raise ValueError("OffgridField is defined by a tuple (type, [axes_size..])")
+
+        dtype, data_dims = field_spec
+
+        return _FieldDescriptor(dtype, [], data_dims)
+
+
 # GTScript builtins: variable annotations
 Field = _FieldDescriptorMaker()
 """Field descriptor."""
 
+OffgridField = _ReadOnlyFieldDescriptorMaker()
```

Comment:
I personally like ConstantField (constant with respect to the grid), but I think not everyone likes it for confusion with non-mutable. Here it both meanings coincide.